BrainPortal/app/models/demand.rb

```diff
@@ -0,0 +1,200 @@
+                        :city, :province, :country, :confirm_token
+
+  validates             :login, :length => { :minimum => 3, :maximum => 20 },     :allow_blank => true
+  validates             :login, :format => { :with => /^[a-zA-Z][a-zA-Z0-9]+$/ }, :allow_blank => true
+
+  validates             :email, :format => { :with => /^(\w[\w\-\.]*)@(\w[\w\-]*\.)+[a-z]{2,}$|^\w+@localhost$/i }
+
+  #validates_length_of   :postal_code, :within => 5..10
+
+  OK_COUNTRIES_LIST = [
+   "Afganistan", "Albania", "ALGERIA",  "ANDORRA",   "Argentina",   "Australia",  "austria", "Azerbaijan",
+   "Belarus",   "Belgique", "Belgium",    "Bolivia",   "Brasil", "Brazil",  "Brunei", "Bulgaria",
+   "CA",  "Cambodia", "CAMEROON", "Canada",     "Chile", "China",  "Colombia", "Costa Rica", "Cuba",   "Cyprus", "Czech Republic",
+   "Danmark",   "Denmark", "Deutschland",   "Dominican Republic",  "Dutch Caribbean",
+   "ecuador", "egypt", "el salvador", "England", "Estonia", "Ethiopia",
+   "Finland",  "France",  "Georgia", "Germany", "Greece",  "Guatemala",  "holland", "Honduras", "Hong Kong",   "Hungary",
+   "Iceland", "India", "Indonesia",  "Iran", "Iraq", "Ireland",  "Israel", "Italia", "Italy",
+   "Japan",  "Kazakhstan", "Kenya",  "Korea",  "Kuwait",  "Latvia", "Lebanon",  "Liechtenstein",  "Lithuania",  "Luxembourg",
+   "Macau", "Madrid",  "Malaysia", "mali", "Malta",  "maroc", "Martinique",   "Mexico",  "Moldova", "montevideo",   "Morocco",
+   "Nederland",   "Nepal", "Netherland", "Netherlands",      "New Zealand", "Nicaragua",  "Nigeria",  "Northern Ireland", "Norway",
+   "Oman",  "Pakistan",  "Paraguay",   "Peru",  "Philippines", "Poland", "Polska",  "Portugal",  "Puerto Rico",  "Qatar",
+   "Republic of Korea",  "Romania",   "Russia",   "Russian Federation", "Rwanda",
+   "Saudi Arabia",  "Schweiz", "Scotland",  "Serbia",   "Singapore",  "Slovakia", "Slovenia", "South Africa", "south korea", "Spain", "sri lanka",  "St. Maarten",  "Suisse", "Suriname", "Sweden", "Switzerland",
+   "Taiwan",  "Thailand",  "The Netherlands", "Trinidad and Tobago",   "Tunisia", "turkey",
+   "U. S. A.", "U.K.", "U.S.", "U.S.A.", "UAE", "UK", "Ukraine",  "United Arab Emirates", "United Kingdom",    "United States", "United States of America",  "Uruguay", "US", "USA",
+   "Venezuela",  "Vietnam",  "Wales"
+  ]
+  OK_COUNTRIES_HASH = {}; OK_COUNTRIES_LIST.each { |c| OK_COUNTRIES_HASH[c.downcase]=true }
+
+
+  def strip_blanks #:nodoc:
+    [
+      :title, :first, :middle, :last,
+      :institution, :department, :position, :email,
+      :street1, :street2, :city, :province, :country, :postal_code,
+      :login, :comment
+    ].each do |att|
+      val = read_attribute(att) || ""
+      write_attribute(att, val.strip)
+    end
+    self.login = (self.login.presence || "").downcase
+    true
+  end
+
+  # Token inserted in email to new user to confirm their address
+  def generate_token
+    tok = ""
+    20.times { c=sprintf("%d",rand(10)); tok += c }
```

Comment:
Make the token start with a non-numeric. For MySQL security, as shown in the phrack article.